tt_eager/tt_dnn/op_library/sharded/sharded_op.hpp

```diff
@@ -141,6 +181,21 @@ inline Tensor sharded_to_interleaved(
         .at(0);
 }
 
+inline Tensor reshard(const Tensor &input_tensor, const MemoryConfig &output_mem_config) {
+    TT_FATAL(input_tensor.shard_spec().has_value());
+    TT_FATAL(output_mem_config.is_sharded());
+
+    return operation::run(
+               Reshard{
+                   .input_mem_config = input_tensor.memory_config(),
```

Comment:
Any reason why input_mem_config is an attribute if input tensor is an input that we take memconfig from?

Pr is this for combining with the other struct?

Reply:
Ya this was meant to combine later but I think I will remove it for now, and when combining I can add it back.